gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py

```diff
@@ -634,183 +933,93 @@ def main(args):
-                x,
-            )
-
-        freq_hts = [freq_hts[x] for x in ["release", "pop_diff", "added", "subtracted"]]
-        ht = hl.Table.multi_way_zip_join(freq_hts, "ann_array", "global_array")
-        ht = ht.checkpoint(new_temp_file("join", extension="ht"), overwrite=True)
-
-        logger.info("Merging AFs from diff pop samples and added samples...")
-        global_array = ht.index_globals().global_array
-        freq_expr, freq_meta_expr, sample_count_expr = merge_freq_arrays(
-            [ht.ann_array[i].freq for i in range(3)],
-            [global_array[i].freq_meta for i in range(3)],
-            operation="sum",
-            count_arrays={
-                "freq_meta_sample_count": [
-                    global_array[i].freq_meta_sample_count for i in range(3)
-                ]
-            },
-        )
-        ht = ht.annotate(freq=freq_expr)
-        ht = ht.annotate_globals(
-            freq_meta=freq_meta_expr,
-            freq_meta_sample_count=sample_count_expr["freq_meta_sample_count"],
-        )
-        ht = ht.checkpoint(new_temp_file("added", extension="ht"), overwrite=True)
-
-        logger.info("Merging AFs from subtracted samples...")
-        freq_expr, freq_meta_expr, sample_count_expr = merge_freq_arrays(
-            [ht.freq, ht.ann_array[3].freq],
-            [ht.index_globals().freq_meta, global_array[3].freq_meta],
-            operation="diff",
-            set_negatives_to_zero=False,
-            count_arrays={
-                "freq_meta_sample_count": [
-                    ht.index_globals().freq_meta_sample_count,
-                    global_array[3].freq_meta_sample_count,
-                ]
-            },
+        group_membership_ht = get_group_membership_ht_for_an(v3_vds.variant_data.cols())
+        group_membership_ht = group_membership_ht.checkpoint(
+            new_temp_file("group_membership_all", "ht")
         )
-        ht = ht.select(freq=freq_expr)
-        logger.info("Making freq_index_dict from freq_meta...")
-        ht = ht.select_globals(
-            freq_meta=freq_meta_expr,
-            freq_meta_sample_count=sample_count_expr["freq_meta_sample_count"],
-            freq_index_dict=make_freq_index_dict_from_meta(hl.literal(freq_meta_expr)),
-        )
-        ht = ht.checkpoint(new_temp_file("subtracted", extension="ht"), overwrite=True)
 
-        logger.info(
-            "Set Y-variant frequency callstats for female-specific "
-            "metrics to missing structs..."
-        )
-        ht = ht.annotate(freq=set_female_y_metrics_to_na_expr(ht))
+        logger.info("Computing the AN HT for new v4 genomes variants...")
+        ht = compute_an_by_group_membership(v3_vds, group_membership_ht, ht)
+        ht.write(res.v3_release_an_ht.path, overwrite=overwrite)
+
+    # TODO: Look at gene on a sex chromosome, to make sure AN works there too.
+    # TODO: Confirm the sample counts on the AN HT match the sample counts created by
```

Comment:
Is it normal that for 'controls_and_biobanks', I only got 'adj' and 'raw', strata by 'sex', by 'pop' but nothing by 'sex & pop' in the test?

Reply:
nope, it should have sex & pop too

Reply:
Okay, problem solved! It was correct.

Reply:
Yay